src/video_core/texture_cache/image.cpp

```diff
@@ -491,13 +550,161 @@ void Image::CopyMip(const Image& src_image, u32 mip, u32 slice) {
+                .aspectMask = vk::ImageAspectFlagBits::eColor,
+                .mipLevel = 0,
+                .baseArrayLayer = mrt1_range.base.layer,
+                .layerCount = mrt1_range.extent.layers,
+            },
+            .dstOffset = {0, 0, 0},
+            .extent = {info.size.width, info.size.height, 1},
+        };
+        scheduler->CommandBuffer().copyImage(src_image.GetImage(),
+                                             vk::ImageLayout::eTransferSrcOptimal, GetImage(),
+                                             vk::ImageLayout::eTransferDstOptimal, region);
+    } else {
+        const vk::ImageResolve region = {
+            .srcSubresource{
+                .aspectMask = vk::ImageAspectFlagBits::eColor,
+                .mipLevel = 0,
+                .baseArrayLayer = mrt0_range.base.layer,
+                .layerCount = mrt0_range.extent.layers,
+            },
+            .srcOffset = {0, 0, 0},
+            .dstSubresource{
+                .aspectMask = vk::ImageAspectFlagBits::eColor,
+                .mipLevel = 0,
+                .baseArrayLayer = mrt1_range.base.layer,
+                .layerCount = mrt1_range.extent.layers,
+            },
+            .dstOffset = {0, 0, 0},
+            .extent = {info.size.width, info.size.height, 1},
+        };
+        scheduler->CommandBuffer().resolveImage(src_image.GetImage(),
+                                                vk::ImageLayout::eTransferSrcOptimal, GetImage(),
+                                                vk::ImageLayout::eTransferDstOptimal, region);
+    }
+
+    flags |= VideoCore::ImageFlagBits::GpuModified;
+    flags &= ~VideoCore::ImageFlagBits::Dirty;
+}
+
+void Image::Clear(const vk::ClearValue& clear_value, const VideoCore::SubresourceRange& range) {
+    const vk::ImageSubresourceRange vk_range = {
+        .aspectMask = vk::ImageAspectFlagBits::eColor,
+        .baseMipLevel = range.base.level,
+        .levelCount = range.extent.levels,
+        .baseArrayLayer = range.base.layer,
+        .layerCount = range.extent.layers,
+    };
+    scheduler->EndRendering();
+    Transit(vk::ImageLayout::eTransferDstOptimal, vk::AccessFlagBits2::eTransferWrite, {});
+    const auto cmdbuf = scheduler->CommandBuffer();
+    cmdbuf.clearColorImage(GetImage(), vk::ImageLayout::eTransferDstOptimal, clear_value.color,
+                           vk_range);
+}
+
+void Image::SetBackingSamples(u32 num_samples, bool copy_backing) {
+    if (!backing || backing->num_samples == num_samples) {
+        return;
+    }
+    ASSERT_MSG(!info.props.is_depth, "Swapping samples is only valid for color images");
+    BackingImage* new_backing;
+    auto it = std::ranges::find(backing_images, num_samples, &BackingImage::num_samples);
+    if (it == backing_images.end()) {
+        auto new_image_ci = backing->image.image_ci;
+        new_image_ci.samples =
+            LiverpoolToVK::NumSamples(num_samples, instance->GetColorSampleCounts());
```

Comment:
```suggestion
            LiverpoolToVK::NumSamples(num_samples, supported_samples);
```

Image has `supported_samples` which records the supported sample count queried from format properties.